onedal/basic_statistics/basic_statistics.py

```diff
@@ -63,59 +62,67 @@ def _get_onedal_params(self, is_csr, dtype=np.float32):
             "result_option": options,
         }
 
-    def _compute_raw(
-        self, data_table, weights_table, module, policy, dtype=np.float32, is_csr=False
-    ):
-        params = self._get_onedal_params(is_csr, dtype)
-
-        result = module.train(policy, params, data_table, weights_table)
 
-        options = self._get_result_options(self.options)
-        options = options.split("|")
+class BasicStatistics(BaseBasicStatistics):
+    """
+    Basic Statistics oneDAL implementation.
+    """
 
-        return {opt: getattr(result, opt) for opt in options}
+    def __init__(self, result_options="all", algorithm="by_default"):
+        super().__init__(result_options, algorithm)
 
-    def _compute(self, data, weights, module, queue):
-        policy = self._get_policy(queue, data, weights)
+    def fit(self, data, sample_weight=None, queue=None):
+        policy = self._get_policy(queue, data, sample_weight)
 
         is_csr = _is_csr(data)
-        if not (data is None) and not is_csr:
-            data = np.asarray(data)
-
-        if not (weights is None):
-            weights = np.asarray(weights)
 
-        data, weights = _convert_to_supported(policy, data, weights)
+        if not (data is None) and not is_csr:
+            data = _check_array(data, ensure_2d=False)
+        if not (sample_weight is None):
+            sample_weight = _check_array(sample_weight, ensure_2d=False)
 
-        data_table, weights_table = to_table(data, weights)
+        data, sample_weight = _convert_to_supported(policy, data, sample_weight)
+        is_single_dim = data.ndim == 1
+        data_table, weights_table = to_table(data, sample_weight)
 
         dtype = data.dtype
-        res = self._compute_raw(data_table, weights_table, module, policy, dtype, is_csr)
+        raw_result = self._compute_raw(data_table, weights_table, policy, dtype, is_csr)
+        for opt, raw_value in raw_result.items():
+            value = from_table(raw_value).ravel()
+            if is_single_dim:
+                setattr(self, opt, value[0])
+            else:
+                setattr(self, opt, value)
 
-        return {k: from_table(v).ravel() for k, v in res.items()}
+        return self
+
+    def compute(self, data, weights=None, queue=None):
+        warnings.warn(
+            "Method `compute` was deprecated in version 2024.3 and will be "
+            "removed in 2024.5. Use `fit` instead."
```

Comment:
Please update in which exact version it will be removed.

Reply:
Also please cover this with test for checking this expected warning for the certain daal4py version. 
This will help us remove the warning from latest versions.